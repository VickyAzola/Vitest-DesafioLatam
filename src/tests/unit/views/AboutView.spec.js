import { describe, it, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import AboutView from '@/views/AboutView.vue'
import { createRouter, createWebHistory } from 'vue-router'

describe('AboutView', () => {
    const wrapper = mount(AboutView)

    it('El HTML del componente se mantiene durante las pruebas', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Probando la existencia de la vista AboutView ', async () => {
        const router = createRouter({
        history: createWebHistory(),
        routes: [
            {
            path: '/about',
            name: 'about',
            component: AboutView
            }
        ]
        })
        router.push({name: 'about'})
        await router.isReady()

        const wrapper = mount(AboutView, {
        global: {
            plugins: [router]
        }
        })
        expect(wrapper.findComponent(AboutView).exists()).toBe(true)
    })
})
