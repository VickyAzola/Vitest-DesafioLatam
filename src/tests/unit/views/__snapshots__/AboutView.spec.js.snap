// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutView > El HTML del componente se mantiene durante las pruebas 1`] = `"<h1>About View</h1>"`;
